src/js/core/player.js

```diff
@@ -727,12 +727,68 @@ function name() {
     const country = playerNames.countries.find(row => row[1] >= cRand)[0];
 
     // First name
-    const fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
-    const firstName = playerNames.first[country].find(row => row[1] >= fnRand)[0];
+    let fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+    let firstName = playerNames.first[country].find(row => row[1] >= fnRand)[0];
+
+    // Middle Name, randomly
+    const middle = random.uniform(0, 1);
+    if (middle < 0.05) {
+        // break loop if there is a country with no simple names
+        let steps = 0;
+        while (/[ .-]/.test(firstName) && steps < 100) {
+            fnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+            firstName = playerNames.first[country].find(row => row[1] >= fnRand)[0];
+            steps = steps + 1;
+        }
+        let mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+        let middleName = playerNames.first[country].find(row => row[1] >= mnRand)[0];
+        steps = 0;
+        while (/[ .-]/.test(middleName) && steps < 100) {
+            mnRand = random.uniform(0, playerNames.first[country][playerNames.first[country].length - 1][1]);
+            middleName = playerNames.first[country].find(row => row[1] >= mnRand)[0];
+            steps = steps + 1;
+        }
+
+        // abbreviation rules:
```

Comment:
As I wrote in the issue thread... does this end up producing weird names too often?